pkg/asset/manifests/infrastructure.go

```diff
@@ -203,6 +215,25 @@ func (i *Infrastructure) Generate(ctx context.Context, dependencies asset.Parent
 			}
 			config.Status.PlatformStatus.GCP.ResourceTags = resourceTags
 		}
+
+		for _, service := range installConfig.Config.Platform.GCP.ServiceEndpoints {
+			var serviceName configv1.GCPServiceEndpointName
+			var ok bool
+			if serviceName, ok = gcpServiceEndpointNameMap[service.Name]; !ok {
+				logrus.Debugf("failed to set GCP service endpoint in infrastructure: %s", service.Name)
+				continue
+			}
```

Comment:
If this happened, seems like we should fail. But this code would not be necessary if we switch our type entirely to use the o/api type.